### PR TITLE
Firefox: 128.0.3 -> 129.0; Firefox-128-ESR: 128.0esr -> 128.1.0esr; Firefox-115-ESR: 115.13.0esr -> 115.14.0esr

### PR DESCRIPTION
**Pull request by mweinelt**
https://www.mozilla.org/en-US/firefox/129.0/releasenotes/
https://www.mozilla.org/en-US/firefox/128.1.0/releasenotes/
https://www.mozilla.org/en-US/firefox/115.14.0/releasenotes/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/browsers/firefox/packages.nix [hunk 1]
```diff
@@ -96,14 +96,14 @@
 
   firefox-esr-128 = buildMozillaMach rec {
     pname = "firefox";
-    version = "128.0esr";
+    version = "128.1.0esr";
```
Comment (kirillrdy) [inline pkgs/applications/networking/browsers/firefox/packages.nix:99]:
```
error: builder for '/nix/store/mcyiqd84495bc8yphpn7h5vllw343qkv-firefox-128.1.0esr.source.tar.xz.drv' failed with exit code 1;
       last 10 log lines:
       >  38  535M   38  207M    0     0  6051k      0  0:01:30  0:00:35  0:00:55 6198k
       > curl: (92) HTTP/2 stream 1 was not closed cleanly: INTERNAL_ERROR (err 2)
       >
       > trying https://archive.mozilla.org/pub/firefox/releases/128.1.0esr/source/firefox-128.1.0esr.source.tar.xz
       > ** Resuming transfer from byte position 217279485
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >  75  328M   75  248M    0     0  8310k      0  0:00:40  0:00:30  0:00:10 8625k
       > curl: (92) HTTP/2 stream 1 was not closed cleanly: INTERNAL_ERROR (err 2)
       > error: cannot download firefox-128.1.0esr.source.tar.xz from any mirror
       For full logs, run 'nix log /nix/store/mcyiqd84495bc8yphpn7h5vllw343qkv-firefox-128.1.0esr.source.tar.xz.drv'.
```

maybe it's just my part of the world ?

Comment (mweinelt) [reply pkgs/applications/networking/browsers/firefox/packages.nix:99]: I think so. Works from here.

Comment (kirillrdy) [reply pkgs/applications/networking/browsers/firefox/packages.nix:99]: Well no need to block because of me

**Comment by mweinelt** [comment]
Build and tested (128/129 only) the source builds on aarch64-linux and x86_64-linux.

**Pull request merged**